docs/architecture.md

```diff
@@ -79,41 +81,201 @@ These components are maintained in the `llm-d-inference-scheduler` repository an
 
 ---
 
-## Scorers & Configuration
+## Configuration
+
+The set of lifecycle hooks (plugins) that are used by the inference scheduler is determined by how 
+it is configured. The configuration is in the form of YAML text, which can either be in a file or
+specified in-line as a parameter. The configuration defines the set of plugins to be instantiated along with their parameters. Each plugin is also given a name, enabling the same plugin type to be instantiated
+multiple times, if needed. Also defined is a set of SchedulingProfiles, which determine the set of
+plugins to be used when scheduling a request. The set of plugins instantiated must also include a
+Profile Handler, which determines which SchedulingProfiles will be used for a particular request.
+
+The configuration text has the following form:
+```yaml
+apiVersion: inference.networking.x-k8s.io/v1alpha1
+kind: EndpointPickerConfig
+plugins:
+- ....
+- ....
+schedulingProfiles:
+- ....
+- ....
+```
 
-| Scorer           | Description                                | Env Vars |
-|------------------|--------------------------------------------|----------|
-| Session-aware    | Prefers pods from same session             | `ENABLE_SESSION_AWARE_SCORER`, `SESSION_AWARE_SCORER_WEIGHT`, `PREFILL_ENABLE_SESSION_AWARE_SCORER`, `PREFILL_SESSION_AWARE_SCORER_WEIGHT` |
-| Prefix-aware     | Scores based on prompt prefix history;<br>lightweight but may not reflect actual KV-cache state | `ENABLE_PREFIX_AWARE_SCORER`, `PREFIX_AWARE_SCORER_WEIGHT`, `PREFILL_ENABLE_PREFIX_AWARE_SCORER`, `PREFILL_PREFIX_AWARE_SCORER_WEIGHT`, `PREFIX_CACHE_HASH_BLOCK_SIZE`, `PREFIX_CACHE_LRU_CAPACITY_PER_SERVER`, `PREFIX_CACHE_MAX_PREFIX_BLOCKS`|
-| KVCache-aware    | Scores based on real KV-cache state on vLLM;<br>more accurate but requires extra computation and cycles to track the current cache state | `ENABLE_KVCACHE_AWARE_SCORER`, `KVCACHE_INDEXER_REDIS_ADDR`, `PREFILL_ENABLE_KVCACHE_AWARE_SCORER`, `PREFILL_KVCACHE_INDEXER_REDIS_ADDR`, `HF_TOKEN`, `KVCACHE_INDEXER_REDIS_ADDR` |
-| Load-aware       | Avoids busy pods                           | `ENABLE_LOAD_AWARE_SCORER`, `LOAD_AWARE_SCORER_WEIGHT`, `PREFILL_ENABLE_LOAD_AWARE_SCORER`, `PREFILL_LOAD_AWARE_SCORER_WEIGHT` |
+The first two lines of the configuration are constant and must appear as is.
 
-### Prefill / Decode Configuration
+The plugins section defines the set of plugins that will be instantiated and their parameters. Each entry in this section
+has the following form:
+```yaml
+- name: aName
+  type: a-type
+  parameters:
+    parm1: val1
+    parm2: val2
+```
+The fields in a plugin entry are:
+- *name* which is optional, provides a name by which the plugin instance can be referenced. If this
+field is omitted, the plugin's type will be used as its name.<br>
+- *type* specifies the type of the plugin to be instantiated.<br>
```

Comment:
do we need `<br>`?

Reply:
have you tried double space at the end of the line?
I think this should give you the same behavior, just without html tags.

Reply:
I don't like trailing spaces, which are unseen and easily get lost